### DISCUSSION TITLE
Check for incorrect argument data type or missing argument in call to function 'sdpvar'

### DISCUSSION THREAD

**Discussion by emonhossainraihan** [yalmip/YALMIP#1099] [Problems using YALMIP] [answered]
I got error when using mesh3DToYalmip function. 

```matlab
prm.debug = true;
prm.visualize = true;

prm.init.method = 'solve_unconstrained';
prm.constraints.C = 1000; 
prm.solve.solver = 'mosek'; %solver of choice
prm.solve.max_iter = 1000;
prm.solve.thresh_obj = 1e-6;
prm.solve.thresh_Y = 1e-6;
prm.solve.yalmip_prm = sdpsettings('solver',prm.solve.solver,'verbose',0);
prm.solve.yalmip_prm.quadprog.Algorithm = 'interior-point-convex';

MAX_ITER = 150;

%% load mesh

mesh_path = '.\models\';
meshname = 'cathead.obj';

OBJ = read_wobj([mesh_path meshname]);
X = OBJ.vertices;
if(length(OBJ.objects)==4)
    tri = OBJ.objects(4).data.vertices;%for meshlab objs
else
    tri = OBJ.objects.data.vertices;%for standard objs
end
        
TR = triangulation(tri,X);

FBtri = freeBoundary(TR);
anchor_ind = FBtri(:,1);
tet = linspace(0,2*pi-eps,length(anchor_ind)+1);
anchor_coords = [cos(tet') sin(tet')];
anchor_coords(end,:)=[];


%% initialize variables
[Y, Faces] = mesh3DToYalmip(X,tri,prm);
```

Error log:

```
Check for incorrect argument data type or missing argument in call to function 'sdpvar'.

Error in mesh3DToYalmip (line 5)
Y = sdpvar(size(X,1), 2);%always map to the plane

Error in demo (line 73)
[Y, Faces] = mesh3DToYalmip(X,tri,prm);
```

It seems sdpvar is not working properly. 

Version:

```
>> yalmip('version')

ans =

    '20210331'
```

**Comment by johanlofberg** [comment, marked as answer]
mesh3DToYalmip is not a YALMIP function, so you would have to contact the authors of that function.

**Reply by johanlofberg** [reply]
And you say "it seems". Have you actually checked if YALMIP is installed properly, i.e., made your own call to sdpvar to see that it works

**Reply by emonhossainraihan** [reply]
Sorry, I guess mesh3DToYalmip is a YALMIP function. But as you suggested, I try to see sdpvar working or not. From [here](https://yalmip.github.io/command/sdpvar/) I copy and paste some code to test, but the same error was shown. 

```matlab
>> X = sdpvar(3,3,3,3,'full')
Check for incorrect argument data type or missing argument in call to function 'sdpvar'.
```

I correctly add paths of YALMIP.  

![Screenshot 2022-03-21 175332](https://user-images.githubusercontent.com/34622219/159255964-82935f70-4d1d-4f2a-abc2-8f3df37a3186.png)

How could I check the installation was done properly? @johanlofberg

**Reply by johanlofberg** [reply]
You've got some weird broken version of sdpvar,i.e. which -all sdpvar probably shows something weird

and why have you installed yalmip under another toolbox. sounds like a bad idea

**Reply by johanlofberg** [reply]
and since you don't appear to know what mesh3DToYalmip is (although you must have installed it),  which uses sdpvar, I would not be surprised if you have multiple versions of YALMIP installed or something like that

**Reply by emonhossainraihan** [reply]
The problem is fixed when I change the directory from `toolbox` to other place. Thanks for your response @johanlofberg.
